azurerm/internal/services/monitor/monitor_metric_alert_resource.go

```diff
@@ -352,37 +491,97 @@ func resourceArmMonitorMetricAlertDelete(d *schema.ResourceData, meta interface{
 	return nil
 }
 
-func expandMonitorMetricAlertCriteria(input []interface{}) *insights.MetricAlertSingleResourceMultipleMetricCriteria {
-	criteria := make([]insights.MetricCriteria, 0)
+func expandMonitorMetricAlertCriteria(d *schema.ResourceData) insights.BasicMetricAlertCriteria {
+	switch {
+	case d.Get("criteria").(*schema.Set).Len() != 0:
+		return expandMonitorMetricAlertMetricCriteria(d.Get("criteria").(*schema.Set).List())
+	case d.Get("dynamic_criteria").(*schema.Set).Len() != 0:
+		return expandMonitorMetricAlertDynamicMetricCriteria(d.Get("dynamic_criteria").(*schema.Set).List())
+	case d.Get("webtest_location_availability_criteria").(*schema.Set).Len() != 0:
+		return expandMonitorMetricAlertWebtestLocAvailCriteria(d.Get("webtest_location_availability_criteria").([]interface{}))
+	default:
+		// Guaranteed by schema `AtLeastOne` constraint
+		panic("will never happen")
```

Comment:
this needs to be removed and replaced by an error so we can gracefully fail

Reply:
Sure